mesonbuild/backend/ninjabackend.py

```diff
@@ -700,6 +703,11 @@ def generate_compdb(self):
                           for ext in ['', '_RSP']]
                 rules += [f"{rule}{ext}" for rule in [self.compiler_to_pch_rule_name(compiler)]
                           for ext in ['', '_RSP']]
+                # Add custom MIL link rules to get the files compiled by the TASKING compiler family to MIL files included in the database
+                key = OptionKey('b_tasking_mil_link')
+                if key in compiler.base_options:
+                    rules += [f"{rule}{ext}" for rule in [self.get_compiler_rule_name('tasking_mil_compile', compiler.for_machine)]
+                              for ext in ['', '_RSP']]
```

Comment:
This line is overcomplicated. Please imbricated for loops instead. Not sure why you iterate through a one-item list.